lib/src/components/NxCheckbox/types.ts

```diff
@@ -4,24 +4,34 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import { LabelHTMLAttributes } from 'react';
+import { InputHTMLAttributes, LabelHTMLAttributes, Validator } from 'react';
 import * as PropTypes from 'prop-types';
 
+export type InputAttributesProp =
+  Omit<InputHTMLAttributes<HTMLInputElement>,
+  'disabled' | 'checked' | 'readOnly' | 'onChange'>;
+
 export type Props = Omit<LabelHTMLAttributes<HTMLLabelElement>, 'onChange'> & {
   checkboxId?: string | null;
   onChange?: (() => void) | null;
   isChecked: boolean;
   disabled?: boolean | null;
   overflowTooltip?: boolean | null;
+  inputAttributes?: InputAttributesProp;
+};
+
+// For testing
+export type PropsWithAnyInputAttributes = Props & {
```

Comment:
I don't think you need this; you can get away with a cast in the test like this:
```
    const component = getShallowComponent({
      checkboxId: 'not-garfield',
      disabled: true,
      isChecked: true,
      className: 'label-classname',
      inputAttributes: {
        id: 'garfield',
        name: 'garfield',
        disabled: false,
        className: 'checkbox-classname',
        checked: false
      } as Props['inputAttributes']
    });
```

Reply:
> I don't think you need this; you can get away with a cast in the test like this:
> 
> ```
>     const component = getShallowComponent({
>       checkboxId: 'not-garfield',
>       disabled: true,
>       isChecked: true,
>       className: 'label-classname',
>       inputAttributes: {
>         id: 'garfield',
>         name: 'garfield',
>         disabled: false,
>         className: 'checkbox-classname',
>         checked: false
>       } as Props['inputAttributes']
>     });
> ```

Fixed: d9cf5af435f57153f63578b8eec1a151a520a8cc